### PR TITLE
[renault] API key to configuration parameter and rule update.

### PR DESCRIPTION
**Pull request by dougculnane**
The API key changes once a year apparently... This causes a communication error and a quick API Code update patch.  I have moved the API key from constants to config. This allows it to be updated with the new value without a re-compile.  I have documented this in the README.

I also updated the charge limit rule to better cope with NULL values in the ChargeMode.

Signed-off-by: Doug Culnane <doug@culnane.net>

### CODE REVIEW

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.renault/README.md [hunk 1]
```diff
@@ -88,9 +92,16 @@ sitemap renaultcar label="Renault Car" {
 
 ![Sitemap](doc/sitemap.png)
 
-If you do not have a smart charger and want to limit the charge of the battery you can set up an  active 15 minute charge schedule in the MyRenault App.
-Then create a Dimmer item "RenaultCar_ChargeLimit" and set it to 80% for example. This rule will change the RenaultCar_ChargingMode to schedule_mode when the limit is reached.
-This stops charging after the battery level reaches the charge limit.
+If you want to limit the charge of the car battery to less than 100%, this can be done as follows. 
+ 
+ * Set up an active 15 minute charge schedule in the MyRenault App.
```
[the remark this answers is not in the document]
Comment (dougculnane) [reply bundles/org.openhab.binding.renault/README.md:97]: A 15 minute thing takes 15 minutes...  I think this is correct. (I used to be English...)

I will change to "dummy charge schedule" which is hopefully clearer about what we are trying to create.

File: bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -83,6 +83,11 @@
 				<description>How long to wait for commands to reach car and update to server in seconds.</description>
 				<default>30</default>
 			</parameter>
+			<parameter name="kamereonApiKey" type="text" required="true">
```
[the remark this answers is not in the document]
Comment (dougculnane) [reply bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml:86]: I forgot about that.  Thanks for noticing it.

**Comment by dougculnane** [comment]
@lolodomo Thanks for you Review.  I have pushed changes to resolve the issues.

Merci Doug.

File: bundles/org.openhab.binding.renault/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.renault/README.md]: I should clearly stop commenting on English speaking lol

**Review by lolodomo — APPROVED** [review]
LGTM

File: bundles/org.openhab.binding.renault/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dougculnane) [reply bundles/org.openhab.binding.renault/README.md]: @lolodomo  I do make a lot of typos so keep up the good work. :-)

**Pull request merged**